lib/topic_query/private_message_lists.rb

```diff
@@ -264,5 +275,15 @@ def group
     def user_first_unread_pm_at(user)
       UserStat.where(user: user).pluck_first(:first_unread_pm_at)
     end
+
+    def group_with_messages_ids(user)
+      @group_with_messages_ids ||= {}
```

Comment:
Is this `group_with_messages_ids` method called in a loop or otherwise with several different users? If not then why do we have to keep the `user.id` as the key of a hash instead of just storing `@group_with_messages_ids_for_user = user.groups.where(has_messages: true).pluck(:id)`? Just curious, I cannot tell from just this diff + file.

Reply:
Actually, I'm constraint by the API design here where we don't have a `acting_user` and `target_user` pattern for this calss. I opted for a hash because I want to err on the safe side since the `target_user` is always passed into the query method calls above. Who knows when someone might call this in a loop.

Reply:
Fair enough!